modules/members.js

```diff
@@ -0,0 +1,103 @@
+// @ts-check
+
+const Augur = require("augurbot-ts"),
+  u = require("../utils/utils"),
+  c = require("../utils/modCommon"),
+  Discord = require("discord.js"),
+  config = require("../config/config.json"),
+  badgeData = require("../utils/badges"),
+  RankInfo = require("../utils/rankInfo"),
+  Jimp = require("jimp");
+
+let font, cardBackground;
+
+/**
+ * Creates a profile card - a PNG that contains some user information in a fun format!
+ * @param {Discord.GuildMember} member The member to create the profile card for.
+ * @returns {Promise<Buffer|undefined>} File-like object to attach to your response.
+ */
+async function makeProfileCard(member) {
+  try {
+    const card = cardBackground.clone();
+
+    const members = member.guild.members.cache;
+    const rank = await u.db.user.getRank(member.id, members);
+    const badges = await badgeData(member.roles.cache);
+
+    const avatar = await Jimp.read(member.displayAvatarURL({ size: 64, extension: "png" }));
+
+    card.blit(avatar, 8, 8)
+      // eslint-disable-next-line no-control-regex
+      .print(font, 80, 8, member.displayName.replace(/[^\x00-\x7F]/g, ""), 212)
+      .print(font, 80, 28, "Joined: " + (member.joinedAt ? u.moment(member.joinedAt).format("MMMM D, YYYY") : "???"), 212);
+
+    const rankOffset = (rank ? 168 : 80);
+    if (rank) {
+      const level = RankInfo.level(rank.totalXP);
+      card.print(font, 8, 80, `Current Level: ${level} (${rank.totalXP.toLocaleString()} XP)`, 284)
+        .print(font, 8, 100, `Next Level: ${RankInfo.minXp(level + 1).toLocaleString()} XP`, 284)
+        .print(font, 8, 128, `Season Rank: ${rank.rank.season}/${member.guild.memberCount}`, 138)
+        .print(font, 154, 128, `Lifetime Rank: ${rank.rank.lifetime}/${member.guild.memberCount}`, 138);
+    }
+
+    for (let i = 0; i < badges.length; i++) {
+      const badge = await Jimp.read('./media/badges/' + badges[i].image);
+      card.blit(badge.resize(61, 61), 10 + (73 * (i % 4)), rankOffset + (73 * Math.floor(i / 4)));
+    }
```

Comment:
this is fine as is, im just going to throw this out there in case you think it might be more readable.
```suggestion
    badges.map((b, i) => {
      const badge= await Jimp.read(`./media/badges/${b.image}`)
      card.blit(badge.resize(61, 61), 10 + (73 * (i % 4)), rankOffset + (73 * Math.floor(i / 4)));
    });
```

Reply:
Added, which helps with concurrency too. Thanks!